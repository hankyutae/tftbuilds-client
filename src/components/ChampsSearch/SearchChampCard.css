.champ-short-display{
  position:relative;
  height:110px;
  min-width:200px;
  margin: 5px 10px;
  background-color:rgb(63, 69, 122);
  color: white;
  text-align: left;
  font-size:100%;
}

/* @media only screen and (max-width:440px){
  .champ-short-display{
    width:100%;
  }
} */

.champ-long-display{
  position:relative;
  min-height:500px;
  margin: 15px 10px;
  background-color:rgba(6, 15, 102, 0.74);
  color: white;
}

.champ-icon-box{
  font-size:100%;
  padding:10px;
  padding-bottom:47px;
}

.champ-icon-box ul{
  display:inline-block;
  margin-top:0;
  margin-bottom:0;
}

.champ-icon{
  float:left;
  width:50px;
}
.champ-name-box{
  position:absolute; 
  left:0;
  right:0;
  bottom:0;
  background-color:rgb(117, 68, 8);
  font-size:100%;
  padding:7px 15px;
}
.champ-name-box-name{
  float:left;
}
.champ-name-box-cost{
  float:right;
}

.coins-white{
  display:inline-block;
  width:10px;
  height:10px;/* 
  background-size: cover; */
  background-repeat: no-repeat;/* 
  background-position: 50% 50%; */
  background-image:url('../../img/coins_gold.png');
}

.for-collapsing-margin{
  height:15px;
  width:100%;
}

.add-to-current-build{
  position:absolute;
  /* top:60px;
  right:15px; *//* 
  bottom:4px;
  right:45px; */
  top:60px;
  left:15px;
  font-size:100%;
  color:red;
  z-index:10;
}

/* .expand-collapse{
  position:absolute;
  top:10px;
  right:10px;
  font-size:100%;
} */

.champ-icon-in-long{
  width:114px;
  margin:auto;
}

.trait-descriptions-box{
  text-align:left;
}
.padding-top{
  padding-top:1em;
}
.trait-description{
  font-size:95%;
  margin:10px 5%;
  padding:10px;
  border:1px solid burlywood;
}
.champ-stats{
  font-size:95%;
  margin:10px 5%;
  padding:10px;
  border:1px solid burlywood;
  text-align:left;
}
.champ-stats ul{
  list-style-type:none;
  padding-left:10px;
}