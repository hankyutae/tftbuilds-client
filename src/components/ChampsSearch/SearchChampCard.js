import React from 'react';
import ImgLink from '../../services/create-img-link-service';
import './SearchChampCard.css';
//import ChampsSearch from './ChampsSearch';

class SearchChampCard extends React.Component {
  render() {
    if (this.props.locale === 'build-page') {
      return (
        <div className="champ-short-display button-cursor" onClick={(e) => {
          e.stopPropagation();
          let champ = { ...this.props.champ };
          champ.stars = 1;
          champ.items = [];
          this.props.handleAdd(champ);
        }}>
          <div className="champ-icon-box">
            <img className={`build-champ-icon cost${this.props.champ.cost}`} src={ImgLink.createLink(this.props.champ.splash)} alt={`${this.props.champ.name} icon`} />
            <div className='search-champ-icon-box-traits-box'>
              {this.props.champ.traits.map((trait) => {
                return <div className='search-champ-icon-box-trait' key={trait}> {trait} </div>
              })}
            </div>

          </div>
          <div className="champ-name-box">
            <div className="champ-name-box-name">
              {this.props.champ.name}
            </div>

            <div className="champ-name-box-cost">
              <div className="coins-white" alt='gold coins' />
              {this.props.champ.cost}
            </div>

          </div>
        </div >
      );

    }

  }

}

export default SearchChampCard;