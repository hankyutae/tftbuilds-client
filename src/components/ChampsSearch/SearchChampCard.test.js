import React from 'react';
import ReactDOM from 'react-dom';
import SearchChampCard from './SearchChampCard';
import champions from '../TestHelpers/champions';
import renderer from "react-test-renderer";

let champ = champions[Object.keys(champions)[0]];

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
      <SearchChampCard champ={champ} locale={'build-page'}/>
    , div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <SearchChampCard champ={champ} locale={'build-page'}/>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});