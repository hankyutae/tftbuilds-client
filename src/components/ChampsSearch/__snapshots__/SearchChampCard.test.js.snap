// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="champ-short-display button-cursor"
  onClick={[Function]}
>
  <div
    className="champ-icon-box"
  >
    <img
      alt="Twisted Fate icon"
      className="build-champ-icon cost2"
      src="http://raw.communitydragon.org/latest/game/assets/ux/tft/championsplashes/tft_twistedfate.png"
    />
    <div
      className="search-champ-icon-box-traits-box"
    >
      <div
        className="search-champ-icon-box-trait"
      >
         
        Pirate
         
      </div>
      <div
        className="search-champ-icon-box-trait"
      >
         
        Sorcerer
         
      </div>
    </div>
  </div>
  <div
    className="champ-name-box"
  >
    <div
      className="champ-name-box-name"
    >
      Twisted Fate
    </div>
    <div
      className="champ-name-box-cost"
    >
      <div
        alt="gold coins"
        className="coins-white"
      />
      2
    </div>
  </div>
</div>
`;
